components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -73,20 +80,41 @@ const hasDecimalsOverflow = (
   return decimalPlaces !== null && decimalPlaces > 0
 }
 
+const getTokenFromParam = (
```

Comment:
Nice to have (for a different PR):

I think it's interesting to return a `BlockchainToken` even if it's not found in the `tokenList`. This allows us to paste an arbitrary contract address in the token selector search box, and use it for swaps.

There's `useGetTokenInfoQuery` that already does this.